core/baseline-core-v1.0-psd01.md

```diff
@@ -2786,113 +2795,722 @@ The document defines storage capability requirements in the following areas:
+* A BPI Subject has appropriate permissions to access the BPI.
+
+Test Steps:
+
+1. Attempt to access the BPI's user interface or API without providing any authentication credentials.
+2. Verify that the BPI redirects the BPI Subject to a SSO login page or displays a message indicating that authentication is required.
+3. Log in using a SSO provider that has been configured for the BPI.
+4. Verify that the BPI grants access to the BPI Subject after successful authentication.
+5. Attempt to access a protected resource or perform a protected action within the BPI.
+6. Verify that the BPI grants access to the BPI Subject if the BPI Subject has appropriate permissions.
+7. Attempt to access a protected resource or perform a protected action within the BPI using an invalid or expired SSO token.
+8. Verify that the BPI denies access to the BPI Subject and provides an appropriate error message.
+9. Attempt to log out of the BPI.
+10. Verify that the BPI logs out the BPI Subject from the SSO provider and revokes the BPI Subject's BPI session.
+
+Test Passing Criteria: The test will pass if
+
+* The BPI redirects the BPI Subject to a SSO login page or displays a message indicating that authentication is required.
+* Successful login using a SSO provider grants access to the BPI.
+* Access to protected resources or actions within the BPI is granted only to BPI Subjects with appropriate permissions.
+* Invalid or expired SSO tokens result in access being denied and an appropriate error message being displayed.
+* Logging out of the BPI logs the BPI Subject out of the SSO provider and revokes the BPI Subject's BPI session.
+
+
 #### **[R286]**	
 BPI Storage MUST support multi-factor authentication (MFA)
 
 *See the link here for the NIST definition adopted in this document [MFA](https://csrc.nist.gov/glossary/term/mfa).*
 
+[[R286]](#r286) Testability: 
+
+Preconditions:
+
+* A BPI test environment is installed and configured properly.
+* The BPI storage component is designed to support MFA.
+* A BPI Subject has appropriate permissions to access the BPI.
+
+Test Steps:
+
+1. Attempt to access the BPI's user interface or API without providing any authentication credentials.
+2. Verify that the BPI redirects the BPI Subject to a login page or displays a message indicating that authentication is required.
+3. Enter valid username and password credentials to log in to the BPI.
+4. Verify that the BPI grants access to the BPI Subject after successful authentication with username and password.
+5. Attempt to access a protected resource or perform a protected action within the BPI.
+6. Verify that the BPI grants access to the BPI Subject if the BPI Subject has appropriate permissions.
+7. Attempt to enable MFA for the BPI Subject account used to log in to the BPI.
+8. Verify that the BPI supports multiple factors for MFA, such as SMS, email, or an authenticator app.
+9. Attempt to log out and log back in to the BPI using only the username and password credentials.
```

Comment:
As written, I don't think one of the factors needs to be username+password. Which would mean this step isn't appropriate.

Reply:
removed and reworded. steps still required.